### PR TITLE
Final refactor

### PR DESCRIPTION
**Pull request by zacisaacson**

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ def initialize(contents)
 
   def add_item(item)
     @contents[item] = 0 if !@contents[item]
-    @contents[item] += 1
+    @contents[item] += 1 if !limit_reached?(item)
```
Comment (johnktravers) [inline app/models/cart.rb:10]: Very nice limitation

File: app/models/order.rb [hunk 2]
```diff
@@ -10,4 +10,8 @@ class Order <ApplicationRecord
   def grandtotal
     item_orders.sum('price * quantity')
   end
+
+  def total_item_quantity
+    item_orders.sum(:quantity)
+  end
```
Comment (johnktravers) [inline app/models/order.rb:16]: Love the helper methods

File: config/routes.rb [hunk 3]
```diff
@@ -1,40 +1,25 @@
...
-  delete '/items/:id', to: 'items#destroy'
+  resources :merchants do
+    resources :items, only: [:index, :new, :create]
+  end
 
-  get '/items/:item_id/reviews/new', to: 'reviews#new'
-  post '/items/:item_id/reviews', to: 'reviews#create'
+  resources :items, except: [:new, :create] do
+    resources :reviews, only: [:new, :create]
+  end
```
Comment (johnktravers) [inline config/routes.rb:12]: These resources look great!

File: config/routes.rb [hunk 4]
```diff
@@ -1,40 +1,25 @@
...
-  get '/items/:id', to: 'items#show'
-  get '/items/:id/edit', to: 'items#edit'
-  patch '/items/:id', to: 'items#update'
-  get '/merchants/:merchant_id/items', to: 'items#index'
-  get '/merchants/:merchant_id/items/new', to: 'items#new'
-  post '/merchants/:merchant_id/items', to: 'items#create'
-  delete '/items/:id', to: 'items#destroy'
+  resources :merchants do
+    resources :items, only: [:index, :new, :create]
+  end
```
Comment (grwthomps) [inline config/routes.rb:8]: Really nice refactor with resources

File: spec/features/orders/new_spec.rb [hunk 5]
```diff
@@ -34,5 +34,14 @@
...
     end
+
+    # it 'cannot create an order without items', js: true do
+    #   page.evaluate_script('window.history.back()')
+    #
+    #   click_on 'Checkout'
+    #
+    #   expect(current_path).to eq('/items')
+    #   expect(page).to have_content("Please add something to your cart to place an order")
+    # end
```
Comment (johnktravers) [inline spec/features/orders/new_spec.rb:45]: Great thought testing the back button and creating a new order! Maybe we'll get those two lines covered one day

Comment (grwthomps) [inline spec/features/orders/new_spec.rb:45]: Really wish we could figure out how to use this test!

**Review by johnktravers — APPROVED** [review]
Great work! I'm very happy with how this project has turned out

**Review by grwthomps — APPROVED** [review]

**Pull request merged**